plugins/processors/gpuattributes/internal/awsneuron_metric_checker.go

```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package internal
+
+import (
+	"regexp"
+)
+
+type AwsNeuronMetricChecker struct {
+}
+
+func NewAwsNeuronMetricChecker() *AwsNeuronMetricChecker {
+	return &AwsNeuronMetricChecker{}
+}
+
+func (md *AwsNeuronMetricChecker) IsProcessedNeuronMetric(name string) bool {
+	matched, err := regexp.MatchString("^(container|node|pod)_(neuroncore_|neurondevice_).*|^node_neuron_.*", name)
```

Comment:
Can make this regex as a constant. Also, I think this func can be moved into processor since what it does is a simple regex match.

Reply:
its easier to test this way, moved the constant